### PR TITLE
PnP sample refinements

### PR DESCRIPTION
**Pull request by jspaith**
Further cleanup of PNP samples, most importantly returning an empty JSON blob on commands (hub/explorer aren't happy otherwise the way C SDK works).

Also note that many files have been renamed from like `device` to `device_ll`.  This is in anticipation of moving to LL style handles.  I've coded up the LL move but unfortunately git is confused with # of changes and doesn't track the rename, making PR and history unpleasant.  Rename the files for now with plan to get the LL work in soon after this hits master.

### CODE REVIEW

File: iothub_client/samples/pnp/pnp_simple_thermostat/pnp_simple_thermostat.c [hunk 1]
```diff
@@ -192,6 +196,26 @@ static bool BuildMaxMinCommandResponse(unsigned char** response, size_t* respons
     return result;
 }       
 
+//
+// SetEmptyCommandResponse sets the response to be an empty JSON.  IoT Hub needs
+// legal JSON, regardless of error status, so if command implementation did not set this do so here.
+//
+//
```
Comment (danewalton) [inline iothub_client/samples/pnp/pnp_simple_thermostat/pnp_simple_thermostat.c:203]: one too many `//`?

File: iothub_client/samples/pnp/pnp_simple_thermostat/readme.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# PnP Simple Thermostat sample
+
+This directory contains a sample thermostat that implements the model [dtmi:com:example:Thermostat;1](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json).  
+
+This sample is relatively straightforward, only implementing telemetry, properties, and commands *without* using sub-components.  Compare this to the more complex [tpnp_temperature_controller](../pnp_temperature_controller).
```
Comment (danewalton) [inline iothub_client/samples/pnp/pnp_simple_thermostat/readme.md:5]:
```suggestion
This sample is relatively straightforward, only implementing telemetry, properties, and commands *without* using sub-components.  Compare this to the more complex [pnp_temperature_controller](../pnp_temperature_controller).
```

File: iothub_client/samples/pnp/pnp_temperature_controller/readme.md [hunk 3]
```diff
@@ -17,3 +17,21 @@ Note that the individual components are in separate .c and .h files for easier c
...
+
+## Configuring the sample
+
+The `pnp_temperature_controller` uses environment variables to retrieve configuration.  
+
+* If you are using a connection string to authenticate:
+  * set IOTHUB_DEVICE_SECURITY_TYPE="connectionString"
+  * set IOTHUB_DEVICE_CONNECTION_STRING="\<connection string of your device\>"
+
+* If you are using a DPS enrollment groups to authenticate:
```
Comment (danewalton) [inline iothub_client/samples/pnp/pnp_temperature_controller/readme.md:29]:
```suggestion
* If you are using a DPS enrollment group to authenticate:
```

File: iothub_client/samples/pnp/pnp_simple_thermostat/pnp_simple_thermostat.c (hunk 1 above)
Comment (jspaith) [reply iothub_client/samples/pnp/pnp_simple_thermostat/pnp_simple_thermostat.c:203]: fixed.

File: iothub_client/samples/pnp/pnp_simple_thermostat/readme.md (hunk 2 above)
Comment (jspaith) [reply iothub_client/samples/pnp/pnp_simple_thermostat/readme.md:5]: Fixing

File: iothub_client/samples/pnp/pnp_temperature_controller/readme.md (hunk 3 above)
Comment (jspaith) [reply iothub_client/samples/pnp/pnp_temperature_controller/readme.md:29]: Fixed.

**Pull request merged**